### ISSUE TITLE
additional redirections flags

### ISSUE THREAD

**Issue by travisghansen** [containous/traefik#6474]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I've deployed `v2.2` via the new (https://github.com/containous/traefik-helm-chart) helm chart and am testing out various things. I've attempted to put a blanket redirect in place using `entrypoints.web.http.redirections.entrypoint.to=websecure` however it's brought a couple issues:

- I'd like to be able to set the `permanent` value of the middleware to `false`
- By default, the chart binds `websecure` to `8443` so without being able to also set the `port` of the middleware I'm being redirected to the wrong port as the kubernetes `service` is on `443`

I'd suggest adding additional cli args:

- `--entrypoints.web.http.redirections.entrypoint.permanent=`
- `--entrypoints.web.http.redirections.entrypoint.port=`

**Comment by ldez** [comment]
Kind of duplicate of #6448

**Comment by ldez** [comment]
Could you explain why do you want to set `permanent` to false ?

**Comment by travisghansen** [comment]
Not super critical the permanent aspect no, I just generally do temporary to make testing easier and not introduce cache behavior in the browsers.

I'll follow the dupe as well. I haven't looked but opening a port may be hard if the container isn't running as root... haven't looked into that. I assumed it would be possible to tweak those values but they're hard-coded in the chart which feels odd to be changing but we can move the issue to the chart if you feel it's better over there.

What ultimately led to this was (what I felt) was strange behavior of the `kubernetesingress` provider. For each `Ingress` object I have 1 router entry for the `traefik` `web` and `websecure` entry points. However, they are flagged as `tls` so trying to hit anything on port 80 returns a 404. If I curl with https but force port 80 I get expected behavior.

**Comment by travisghansen** [comment]
Continues from my laptop...

I then added the following:

```
     - "--entrypoints.web.http.tls=false"
     - "--entrypoints.web.http.middlewares=adc-system-httpredirect@kubernetescrd"
```

That results in multiple router entries per `Ingress`, 1 with `traefik` and `websecure` and a 2nd with `web` but the `web` entry is also flagged as `tls` so I landed in the same predicament of https traffic expected to arrive on the http port.

If I've got something wrong in the approach just shout out what you think I should try..

**Comment by ldez** [comment]
> What ultimately led to this was (what I felt) was strange behavior of the kubernetesingress provider. For each Ingress object I have 1 router entry for the traefik web and websecure entry points. However, they are flagged as tls so trying to hit anything on port 80 returns a 404. If I curl with https but force port 80 I get expected behavior.

In the v2 the default entrypoints are all entrypoints, so either you have to define the good entry point on each ingress or use the redirection.

I recommend to set explicitly the `websecure` entry point, because it's a way more secure from my point of view.

But it's off-topic, if you want to discuss more please come to the [community forum](https://community.containo.us/)

**Comment by ldez** [comment]
To return to the original subject, unlike the previous issue, the helm-chart is not an edge case.
So I need to think about that.

**Comment by travisghansen** [comment]
Sure, thanks for the consideration.

That's pretty intrusive to force users to add an annotation on each `Ingress`. I'd be happy to use the `redirection` feature if I could get it to redirect to the proper port. I'll force update some more of the defaults but it may be something to consider tweaking via adding the args mentioned, or updating the chart to behave slightly differently as users would probably get pretty confused by the intersection of required params to enable a pretty common scenario.

**Comment by travisghansen** [comment]
Yeah, so this works with the chart:

```
  -  additionalArguments:
     ...
     - "--entrypoints.web.http.redirections.entrypoint.to=websecure"
     ports:
       websecure:
         port: 443
```

Not sure how much of an issue it is to run the container as `root` using privileged ports (not really an issue for me, but may be for others I suppose) but it works for me well enough ATM.

**Comment by travisghansen** [comment]
As an operator I want to be able to optionally **force** this behavior as well. Unless I missing something this currently seems impossible. From what I can tell that would involve being able to set the priority on the `web-to-websecure@internal` rule arbitrarily high.

Use-case: for certain ingress classes I do not want to allow any insecure connections. From what I can tell if an `Ingress` is created without the `tls` block **AND** the auto-generated rule has more characters than ```HostRegexp(`{host:.+}`)``` the request would be served (even when using the `--entrypoints.web.http.redirections.entrypoint.to=websecure` option). On other ingresses classes clear could/should be allowed.

**Comment by ldez** [comment]
I'm already working on it https://github.com/ldez/traefik/commit/ba8fac44fa4b07e4b762a2f3624c3827081fd258

**Comment by travisghansen** [comment]
Nice! Any more thought on the `port` setting?

**Comment by traefiker** [comment]
Closed by #6512.
